### DISCUSSION TITLE
Fill bogus data fields in content types pulled from Content Definition Manager

### DISCUSSION THREAD

**Discussion by vengi83644** [OrchardCMS/OrchardCore#11225] [Q&A]
I am using Decoupled CMS and have created some content types and stereotyped as widgets to use in flow parts in many pages.

While creating a page, the content author can select any of these widgets and fill in any data as they need.

I want to publish a page where I can show all these widgets and let the author decide which one to use based on the fields and UI.

I am planning to pull all the content definitions that are stereotyped as widgets and fill in some bogus data and render them one by one with their names.

```csharp

Dictionary<string, ContentItem> widgetContents =  new Dictionary<string, ContentItem>();

var allContentTypes = _contentDefinitionManager.ListTypeDefinitions();            

var widgets = allContentTypes.Where(w =>
                                    w.Settings.ContainsKey("ContentTypeSettings") &&
                                    w.GetSettings<ContentTypeSettings>() != null &&
                                    w.GetSettings<ContentTypeSettings>().Stereotype != null &&
                                    w.GetSettings<ContentTypeSettings>().Stereotype.Equals("widget", StringComparison.OrdinalIgnoreCase));

foreach (var widget in widgets)
{
// I want to create a content item that the widget needs.
ContentItem model;

widgetContents.Add(widget.Name, model);
}

```

Here is my cshtml for a single widget

```HTML
@model ContentItem

<div class="container h2Body">
    <div class="row">
        <div class="pt-3 col-xs-12 col-sm-12 col-md-12 col-lg-12">
            <div class="title text-center">
                @Model.Content.H2Body.Title.Text
            </div>
            <div class="text-center col-xs-12 col-sm-12 col-md-12 col-lg-12 fs-5">
                @Html.Raw(Model.Content.H2Body.Body.Html)
            </div>
        </div>
    </div>
</div>
```

I want to dynamically figure out the Title and Body fields and fill in any data to the Text and HTML respectively.

I don't know how to do these dynamically for all widgets. 

Has anyone tried this?

**Comment by Piedone** [comment]
If I understand correctly, you want to loop through the widget content types, and for each of these, you want to create a new widget content item, and fill out their Title and HtmlBody.

If indeed, then [here](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Controllers/PersonListController.cs#L154) you can find an annotated example of how to create and update a content item programmatically. Instead of `PersonPart` you'd access `TitlePart` and `HtmlBodyPart`. Though for title I think you rather want to display a title of the content item than save something into `TitlePart`, for which you need to assign to `person.DisplayText` in the linked example.

**Reply by vengi83644** [reply]
Yes, you are right. But the thing is I don't know about the parts or fields I have in the widget.

I want to recognize them at runtime and assign values.

**Reply by vengi83644** [reply]
I am having around 45 or more widgets and I don't have their schema in my C# code.

I am able to see the content definition from "_contentDefinitionManager.ListTypeDefinitions();". But how to utilize this definition to create a new Content Item.

**Reply by Piedone** [reply]
If you have no information about what may be in those content items then you can't change their values (unless you think about every possible part and field that may be there), you need to know something.

**Reply by vengi83644** [reply]
Yup, I understand. Let me find other way.
